src/context/mod.rs

```diff
@@ -455,6 +496,61 @@ impl<C: Blockchain> MatchstickInstanceContext<C> {
         Ok(true)
     }
 
+    pub fn assert_data_source_count(
+        &mut self,
+        _gas: &GasCounter,
+        template_name_ptr: AscPtr<AscString>,
+        expected_count_ptr: u32,
+    ) -> Result<bool, HostExportError> {
+        let template_name: String =
+            asc_get(&self.wasm_ctx, template_name_ptr, &GasCounter::new(), 0)?;
+
+        let actual_count = self
+            .templates
+            .get(&template_name)
+            .unwrap_or_else(|| panic!("No template with name '{}' found.", template_name))
+            .len() as u32;
+
+        if actual_count != expected_count_ptr {
```

Comment:
I think this statement will always resolve as `false` because you use the pointer of `expected_count` instead of its value

Reply:
The _ptr is misleading, expected_count it the actual value.